streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java

```diff
@@ -42,7 +42,7 @@
     private final Serde<?> valueSerde;
     private boolean initialized;
     protected ProcessorRecordContext recordContext;
-    protected ProcessorNode<?, ?> currentNode;
+    protected ProcessorNode<?, ?, ?, ?> currentNode;
```

Comment:
As you're about to find out, this kind of thing is the bulk of the changes.

Since I've only converted the ProcessorNode (an internal class) in this PR, you'll see the arguments right now are overwhelmingly wildcards and Object. ProcessorNode is almost exclusively used in the "machine room" parts of Streams where we don't have access to, or any benefit from, the actual generic type parameters of the Processors.

In the follow-on PRs, where I convert the actual Processors to the new API is when we'll start to see the benefits.